src/composables/useHotKey/index.ts

```diff
@@ -124,17 +126,34 @@ export function useHotKey(
 
 	/**
 	 * Validates event key to expected key
-	 * FIXME should support any languages / key codes
 	 *
 	 * @param event keyboard event
 	 * @param key expected key
 	 * @return whether it satisfies expected value or not
 	 */
 	const validateKeyEvent = (event: KeyboardEvent, key: string): boolean => {
-		if (options.caseSensitive) {
-			return event.key === key
+		// If key exactly matches event.key, valid with any caseSensitive option. Do not perform further checks
+		if (event.key === key) {
+			return true
+		}
+
+		const isKeyInLowerCase = key === key.toLowerCase()
+		const isEventKeyInLowerCase = event.key === event.key.toLowerCase()
+
+		// If key and event.key are in different cases, invalid with caseSensitive = true. Do not perform further checks
+		if (options.caseSensitive && isKeyInLowerCase !== isEventKeyInLowerCase) {
+			return false
+		} else if (event.key.toLowerCase() === key.toLowerCase()) {
+			return true
+		}
```

Comment:
```suggestion
		// If key exactly matches event.key, valid with any caseSensitive option. Do not perform further checks
		if (event.key === key) {
			return true
		}

		if (options.caseSensitive) {
			const isKeyInLowerCase = key === key.toLowerCase()
			const isEventKeyInLowerCase = event.key === event.key.toLowerCase()

			if (isKeyInLowerCase !== isEventKeyInLowerCase) {
				return false
			}
			return event.key.toLowerCase() === key.toLowerCase()
		}
```

Otherwise the `options.caseSensitive` is ignored if they do not match.
if `caseSensitive` is set this branch **must** return.

Reply:
It's not ignored, and you missing deriving that way

If caseSensitive is true and cases do not match
  return false

Otherwise they match, sensitive option is fulfilled, and we continue checking

Reply:
Then the whole else if can just be removed

Reply:
It can not be removed, it's comparing `z` with `Z` 👀